core/src/test/java/org/apache/iceberg/actions/TestBinPackStrategy.java

```diff
@@ -138,6 +138,43 @@ public void testGroupingMinInputFilesInvalid() {
         0, Iterables.size(grouped));
   }
 
+  @Test
+  public void testGroupingMinInputFilesAsOne() {
+    RewriteStrategy strategy = defaultBinPack().options(ImmutableMap.of(
+            BinPackStrategy.MIN_INPUT_FILES, Integer.toString(1),
+            BinPackStrategy.MAX_FILE_SIZE_BYTES, Long.toString(3 * MB),
+            RewriteDataFiles.TARGET_FILE_SIZE_BYTES, Long.toString(2 * MB),
+            BinPackStrategy.MIN_FILE_SIZE_BYTES, Long.toString(MB),
+            BinPackStrategy.DELETE_FILE_THRESHOLD, Integer.toString(2)
+    ));
+
+    // single file within a group, with size within threshold and no deletes should be considered as NOOP.
+    Iterable<FileScanTask> testFiles1 = filesOfSize(1);
+    Iterable<List<FileScanTask>> grouped1 = strategy.planFileGroups(testFiles1);
+
+    Assert.assertEquals("Should plan 0 group, as 1 minInputFile with size within threshold and no deletes should be " +
+            "treated as NOOP",
+            0, Iterables.size(grouped1));
+
+    // group with single file with size > targetFileSize should not be planed
+    Iterable<FileScanTask> testFiles2 = filesOfSize(1);
+    Iterable<List<FileScanTask>> grouped2 = strategy.planFileGroups(testFiles2);
```

Comment:
I'm not sure how this is different than the first test? Shouldn't this be multiple files less than target file size and less than minInputFiles?

Reply:
Thanks, you are correct these two are exactly the same .

Added test case where 
- multiple files less than target file size and less than minInputFiles will not be planned 
- multiple files greater than target file size and less than minInputFiles will be planned